common/djangoapps/microsite_configuration/middleware.py

```diff
@@ -83,3 +87,54 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
             response.set_cookie = _set_cookie_wrapper
 
         return response
+
+
+class DatabaseMicrositeMiddleware(MicrositeMiddleware):
+    """
+    Middleware class which will bind configuration information regarding 'microsites' on a per request basis.
+    The actual configuration information is taken from the microsite model in the database
+    """
+
+    def process_request(self, request):
+        """
+        Middleware entry point on every request processing. This will associate a request's domain name
+        with a 'University' and any corresponding microsite configuration information
+        """
+        microsite.clear()
+        domain = request.META.get('HTTP_HOST', None)
+        microsite.set_from_db_by_domain(domain)
+        return None
+
+
+class MicrositeCrossBrandingFilterMiddleware():
+    """
+    Middleware class that prevents a course defined in a branded ORG trough a microsite, to be displayed
+    on a different microsite with a different branding.
+    """
```

Comment:
I'm not sure how this will work when Studio goes to the LMS to preview draft content, because I think we use a hostname of 'preview.edx.org' or something like that. I'll have to investigate a bit more closely.